lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs, isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
+
   let npmTarget;
   let npmArgs;
   let result;
   switch (yarnTarget) {
     case "install":
       npmTarget = "install";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
       break;
     case "add":
       npmTarget = "install";
       const packages = yarnSubCommands.map(transformNarnPackageString);
-      const saveType = yarnArgs.dev || yarnArgs.D ? "--save-dev" : "--save";
+      const saveType = yarnArgs.dev ? "--save-dev" : "--save";
       npmArgs = [saveType, ...packages];
       break;
     case "remove":
       npmTarget = "uninstall";
-      npmArgs = yarnSubCommands;
+      npmArgs = narnSubArgs;
       break;
     case "init":
       npmTarget = "init";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
+      break;
+    case "update":
```

Comment:
Yarn does not have an update command. The interface for using narn is the yarn interface. This should be removed

Reply:
As i said previously, update and upgrade are the same, and if this library is intended to bring all those different implementation in a single command, is worth to connect them.

No one in their sane mind would make a package.json script called "update" just for the sake of making the package incompatible with "npm" and/or "pnpm".

I understood your point on the other scripts, but I have to disagree with this one.

Reply:
I am the owner of this project. The API for narn will follow the API for yarn. We will not enter territory where narn does not follow the yarn CLI, unless it *_fully_* supports npm commands being converted to yarn commands.

I will not merge this until we are following the API described in the readme - that narn follows the yarn syntax exactly